### PR TITLE
Added header on the dashboard page.

### PR DESCRIPTION
**Pull request by zacharyhsia**
I agree with Bryan that there is no explicit reason we use a different URL encoding scheme on dashboard page and I changed it.

Also add **header component** on the dashboard page.

### CODE REVIEW

**Comment by zacharyhsia** [comment]
Hi Bryan, ready to be reviewed! Thanks!

**Comment by bfcrampton** [comment]
Taking a look now!

File: ui/src/app/dashboard/dashboard.component.spec.ts [hunk 1]
```diff
@@ -155,17 +167,17 @@ describe('DashboardComponent', () => {
...
     });
 
     tick();
     fixture.detectChanges();
     tick();
-
   }));
 
   it('should create dashboard', fakeAsync(() => {
+    // debugger;
```
Comment (bfcrampton) [inline ui/src/app/dashboard/dashboard.component.spec.ts]: rm

File: ui/src/app/dashboard/grouped-summary/grouped-summary.component.ts [hunk 2]
```diff
@@ -37,7 +37,10 @@ export class GroupedSummaryComponent {
 
   getCommonUrlParamsMap(entry: AggregationEntry) {
     let map = new Map<string, string[]>();
-    map.set('projectId', [this.activatedRoute.snapshot.queryParams['projectId']]);
+
+    const projectId = URLSearchParamsUtils.unpackURLSearchParams(this.activatedRoute.snapshot.queryParams['q']).extensions['projectId'];
```
Comment (bfcrampton) [inline ui/src/app/dashboard/grouped-summary/grouped-summary.component.ts]: Can you wrap this into two lines its pretty long

File: ui/src/app/projects/projects.component.ts [hunk 3]
```diff
@@ -95,21 +95,20 @@ export class ProjectsComponent implements OnInit {
   }
 
   navigateJobs() {
+    const extras = {
```
Comment (bfcrampton) [inline ui/src/app/projects/projects.component.ts]: nit: name it something like projectQueryParams

Comment (zacharyhsia) [reply ui/src/app/projects/projects.component.ts]: Thanks! This value is of `[NavigationExtra](https://angular.io/api/router/NavigationExtras)` class defined by Angular. Only the `queryParams` is about query. Plus, it is originally called `extra`, which might make more sense here.

File: ui/src/app/dashboard/grouped-summary/grouped-summary.component.ts (hunk 2 above)
Comment (zacharyhsia) [reply ui/src/app/dashboard/grouped-summary/grouped-summary.component.ts]: Fixed!

File: ui/src/app/dashboard/dashboard.component.spec.ts (hunk 1 above)
Comment (zacharyhsia) [reply ui/src/app/dashboard/dashboard.component.spec.ts]: :)

**Pull request merged**